mods/ra/maps/allies-06b/allies06b-AI.lua

```diff
@@ -0,0 +1,248 @@
+  information, see COPYING.
+]]
+
+WTransWays =
+{
+	{ WaterUnloadEntry1.Location, WaterUnload1.Location },
+	{ WaterUnloadEntry2.Location, WaterUnload2.Location },
+	{ WaterUnloadEntry3.Location, WaterUnload3.Location }
+}
+
+WTransUnits =
+{
+	hard = { { "3tnk", "3tnk", "3tnk", "v2rl", "v2rl" }, { "v2rl", "v2rl", "e4", "e4", "3tnk" } },
+	normal = { { "e1", "e1", "3tnk", "3tnk", "v2rl" }, { "e4", "e4", "e4", "e4", "v2rl" } },
+	easy = { { "e1", "e1", "e1", "e2", "e2" }, { "e2", "e2", "3tnk" } }
+}
+
+WTransDelays =
+{
+	easy = 4,
+	normal = 3,
+	hard = 1
+}
+
+BuildDelays =
+{
+	easy = 90,
+	normal = 60,
+	hard = 30
+}
+
+WaterAttacks =
+{
+	easy = 1,
+	normal = 2,
+	hard = 3
+}
+
+WaterAttackTypes =
+{
+	easy = { "ss" },
+	normal = { "ss", "ss" },
+	hard = { "ss", "ss", "ss" }
+}
+
+VehicleTypes = { "v2rl", "3tnk", "3tnk", "3tnk", "3tnk", "harv" }
+
+InfTypes =
+{
+	{ "e1", "e1", "e1", "e1", "e1"},
+	{ "e2", "e2", "e1", "e1", "e1"},
+	{ "e4", "e4", "e4", "e1", "e1"}
+}
+
+AttackRallyPoints =
+{
+	{ SovietSideAttack1.Location, SovietBaseAttack.Location },
+	{ SovietBaseAttack.Location },
+	{ SovietSideAttack2.Location, SovietBaseAttack.Location }
+}
+
+ImportantBuildings = { WarFactory, Airfield1, Airfield2, Radar2, Refinery, SovietConyard }
+SovietAircraftType = { "yak" }
+Yaks = { }
+IdlingUnits = { }
+IdlingTanks = { tank1, tank2, tank3, tank4, tank5, tank6, tank7 }
+IdlingNavalUnits = { }
+
+InitialiseAttack = function()
+	Utils.Do(ImportantBuildings, function(a)
+		Trigger.OnDamaged(a, function()
+			Utils.Do(IdlingTanks, function(unit)
+				if not unit.IsDead then
+					IdleHunt(unit)
+				end
+			end)
+		end)
+		Trigger.OnCapture(a, function()
+			Utils.Do(IdlingTanks, function(unit)
+				if not unit.IsDead then
+					IdleHunt(unit)
+				end
+			end)
+		end)
+	end)
+end
+
+Attack = 0
+ProduceInfantry = function()
+	if SovietBarracks.IsDead or SovietBarracks.Owner ~= ussr then
+		return
+	end
+
+	Attack = Attack + 1
+	local toBuild = Utils.Random(InfTypes)
+	ussr.Build(toBuild, function(units)
+		if Attack == 2 and not AttackTank1.IsDead then
+			units[#units + 1] = AttackTank1
+		elseif Attack == 4 and not AttackTank2.IsDead then
+			units[#units + 1] = AttackTank2
+		end
+
+		SendAttack(units, Utils.Random(AttackRallyPoints))
+		Trigger.AfterDelay(DateTime.Seconds(BuildDelays), ProduceInfantry)
+	end)
+end
+
+ProduceVehicles = function()
+	if WarFactory.IsDead or WarFactory.Owner ~= ussr then
+		return
+	end
+	ussr.Build(VehicleTypes, function(units)
+		Utils.Do(units, function(unit)
+			if unit.Type ~= "harv" then
+				IdlingTanks[#IdlingTanks + 1] = unit
+			end
+		end)
+	end)
+end
+
+ProduceNaval = function()
+	if not shouldProduce and #Utils.Where(Map.ActorsInWorld, function(self) return self.Owner == player and self.Type == "syrd" end) < 1 then
+		Trigger.AfterDelay(DateTime.Minutes(1), ProduceNaval)
+		return
+	end
+
+	shouldProduce = true
+
+	if SubPen.IsDead or SubPen.Owner ~= ussr then
+		return
+	end
```

Comment:
You might want to move this to be at the top of the function, no reason to use the Utils.Where function on ActorsInWorld if the SubPen is dead or converted.